### ISSUE TITLE
speech support for systemd-boot?

### ISSUE THREAD

**Issue by mhussaincov** [systemd/systemd#11702]
**Is your feature request related to a problem? Please describe.**
i'm visually impared
would it be possable to have speech on the systemd boot menu?
so when arrowing up and down boot enteries I would get audio feedback from espeak telling me what boot entery i'm on?
it would help alot 
thanks for considdering my feature request.
Majid Hussain
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Comment by poettering** [comment]
Uh, speech support in a boot loader, that would be a new concept, no?

I figure it should be possible to implement an EFI driver that provides a serial driver and synthesizes everything written to it. This could then be loaded into the sd-boot EFI environment and you should get what you want without modifying sd-boot itself. That said, iirc UEFI doesn't do audio drivers, hence not sure how realistic all that is...

Hence, I sympathize, but I am pretty sure this should be a separate project, i.e. a separate EFI driver.

And we should modify sd-boot to automatically load EFI binaries from a specific dir on disk, so that such a text-to-speech driver could be auto-loaded when it exists.

**Comment by CameronNemo** [comment]
[Existing speech synthesizers, last updated 2018-03-31](http://mielke.cc/brltty/doc/Manual-BRLTTY/English/BRLTTY-9.html).

**Comment by hellcp** [comment]
> Uh, speech support in a boot loader, that would be a new concept, no?

Not exactly, gfxboot has support and implements reading available options via, wait for it, PC speaker: https://github.com/openSUSE/gfxboot
